### ISSUE TITLE
False positive resolve of qualified references for non-modules

### ISSUE THREAD

**Issue by himikof** [intellij-rust/intellij-rust#463]
Consider the following resolve test case:
```rust
enum E<T> {
    X
}

fn main() {
    let _ = E::<caret>T;
}
```
Currently resolve for `RustQualifiedReferenceElement` thinks that `T` is bound at that point, because it reuses the notion of `RustResolveScope` (which should probably be strictly lexical). Lexical resolve scope matches qualified resolve scope only for modules, so they should probably be separated.

Also, enums inject their variants into their (lexical?) resolve scope, therefore the inverse test case:
```rust
enum E {
   X, Y(<caret>X)
}
```
also inappropriately resolves.
